include/boost/dynamic_bitset/detail/dynamic_bitset.hpp

```diff
@@ -85,126 +86,129 @@ namespace boost {
-
-             } while (length);
-         }
-
-         return num;
-     }
-
-
-     // Some platforms have fast popcount operation, that allow us to implement
-     // counting bits much more efficiently
-     //
-     template <typename ValueType>
-     BOOST_FORCEINLINE std::size_t popcount(ValueType value) BOOST_NOEXCEPT
-     {
-         std::size_t num = 0;
-         while (value) {
-             num += count_table<>::table[value & ((1u<<table_width) - 1)];
-             value >>= table_width;
-         }
-         return num;
-     }
+    const unsigned int table_width = 8;
+    template <bool b>
+    const byte_type count_table<b>::table[] =
+    {
+        // Automatically generated by GPTableGen.exe v.1.0
+        //
+        0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5,
+        1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6,
+        1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6,
+        2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7,
+        1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6,
+        2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7,
+        2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7,
+        3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 4, 5, 5, 6, 5, 6, 6, 7, 5, 6, 6, 7, 6, 7, 7, 8
+    };
 
-#if ((defined(BOOST_MSVC) && (BOOST_MSVC >= 1600)) || (defined(__clang__) && defined(__c2__)) || (defined(BOOST_INTEL) && defined(_MSC_VER))) && (defined(_M_IX86) || defined(_M_X64))
 
-/*
-     per https://github.com/boostorg/dynamic_bitset/issues/33
-     this code does not support older cpus properly...
-     it needs to check for cpuid support to avoid undefined behavior
-
-     template <>
-     BOOST_FORCEINLINE std::size_t popcount<unsigned short>(unsigned short value) BOOST_NOEXCEPT
-     {
-         return static_cast<std::size_t>(__popcnt16(value));
-     }
-
-     template <>
-     BOOST_FORCEINLINE std::size_t popcount<unsigned int>(unsigned int value) BOOST_NOEXCEPT
-     {
-         return static_cast<std::size_t>(__popcnt(value));
-     }
-
-#ifdef _M_X64
-     template <>
-     BOOST_FORCEINLINE std::size_t popcount<unsigned __int64>(unsigned __int64 value) BOOST_NOEXCEPT
-     {
-         return static_cast<std::size_t>(__popcnt64(value));
-     }
+    // Some platforms have fast popcount operation, that allow us to implement
+    // counting bits much more efficiently
+    //
+    template <typename ValueType>
+    BOOST_FORCEINLINE std::size_t popcount(ValueType value) BOOST_NOEXCEPT
+    {
+        std::size_t num = 0u;
+        while (value) {
+            num += count_table<>::table[value & ((1u<<table_width) - 1)];
+            value >>= table_width;
```

Comment:
Coverity flagged this issue:

CID 338011 (#1 of 1): Bad bit shift operation (BAD_SHIFT)
2. large_shift: In expression value >>= 8U, right shifting value by more than 7 bits always yields zero. The shift amount is 8.

I believe this is intentional in this case, but I want to double-check with you.

Reply:
1. I believe the original code was the same. The change was in the indentation.
2. The intention of the code is to eventually shift off all set bits, leaving `value` zero. If `value` is less than `int`, it is promoted to `int` before the shift operation, so in no case the shift by 8 bits exceeds the left hand operand capacity. I believe, the error is wrong.

Reply:
I'll mark it as such, thanks.  There's another interesting one in the issues which looks like a real bug.